### PR TITLE
FirmwareControl: resume hash save/load changes

### PR DESCRIPTION
**Pull request by HaseenaSainul**

### CODE REVIEW

File: FirmwareControl/DownloadEngine.h [hunk 1]
```diff
@@ -133,15 +137,27 @@ namespace PluginHost {
...
+                                 _HMAC, Crypto::HASH_SHA256) != 0) {
 
-                status = Core::ERROR_UNAUTHENTICATED;
+                        status = Core::ERROR_UNAUTHENTICATED;
+                    }
+
+                } else if (status == Core::ERROR_UNAVAILABLE) {
+                    _storage.LoadFileInfo();
+                    if (_storage.Position() != 0) {
+                        SaveContext((destination).Hash().CurrentContext());
```
Comment (pwielders) [inline FirmwareControl/DownloadEngine.h:158]: I think the rounding brackk=ets are left over from the static_cast from the past ? I assume this shoudl read:  SaveContext(destination.Hash().CurrentContext());

Comment (HaseenaSainul) [reply FirmwareControl/DownloadEngine.h]: corrected

**Review by pwielders — APPROVED** [review]
You are fast !

**Comment by HaseenaSainul** [comment]
:-)

**Pull request merged**